### PR TITLE
LIBCLOUD-698 rename OpSource Dimension Data. Added Dimension Data driver...

### PR DESCRIPTION
**Pull request by tonybaloney**
As per LIBCLOUD-698, OpSource is now Dimension Data. 
Added an additional driver (complete copy).

### CODE REVIEW

**Comment by Kami** [comment]
Sorry for the delay and thanks. Would it be possible for us to get a test account so we can test out the changes on a real installation?

**Comment by Kami** [comment]
Overall, after quickly glancing over it, the changes look good.

Can you please add some documentation to `docs/compute/drivers/` (see existing driver / provider documentation for an example)?

File: libcloud/compute/drivers/dimensiondata.py [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+    DimensionData API pending operation status class
+        action, requestTime, username, numberOfSteps, updateTime,
+        step.name, step.number, step.percentComplete, failureReason,
+    """
+    def __init__(self, action=None, requestTime=None, userName=None,
+                 numberOfSteps=None, updateTime=None, step_name=None,
+                 step_number=None, step_percentComplete=None,
+                 failureReason=None):
+        self.action = action
+        self.requestTime = requestTime
```
Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py]: Heh, looks like this stylistic thing slipped through when we originally reviewed and merged opsource driver.

Variables should use underscore separated names and not camel case. It seems that now that we are adding a new driver is a good time to fix that.

File: libcloud/compute/drivers/dimensiondata.py [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+    name = 'DimensionData'
+    website = 'http://www.dimensiondata.com/'
+    type = Provider.DIMENSIONDATA
+    features = {'create_node': ['password']}
+
+    def create_node(self, **kwargs):
+        """
+        Create a new DimensionData node
+
+        :keyword    name:   String with a name for this new node (required)
```
Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py:217]: Can we please also fix this and explicitly declare arguments in the method signature?

https://libcloud.readthedocs.org/en/latest/development.html#don-t-abuse-kwargs

(I know you just copied over the code from opsource driver, but if we are adding a new driver, let's fix what we can and improve it).

**Comment by tonybaloney** [comment]
Commit added to pull request with changes on casing, docstrings and kwargs refactoring. Will work on the documentation side now.

https://github.com/DimensionDataCBUSydney/libcloud/commit/7faa26f8b14bb5cbf5ea6f0ce4a829cc2ba4ef27

**Comment by tonybaloney** [comment]
Documentation edits have been contributed back to the branch.

Please let me know if we can move ahead.

File: libcloud/compute/drivers/dimensiondata.py [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    DimensionData node driver.
+    """
+
+    connectionCls = DimensionDataConnection
+    name = 'DimensionData'
+    website = 'http://www.dimensiondata.com/'
+    type = Provider.DIMENSIONDATA
+    features = {'create_node': ['password']}
+
+    def create_node(self, name, image, auth, ex_description,
```
Comment (Kami) [inline libcloud/compute/drivers/dimensiondata.py:212]: A lot better, thanks :)

**Comment by Kami** [comment]
@tonybaloney Thanks, changes look good.

Can you please squash the commits and I will go ahead and merge it into trunk.

**Comment by tonybaloney** [comment]
This is the squashed commit. 
https://github.com/DimensionDataCBUSydney/libcloud/commit/c2b4ba6669851294cb5f76cc82aa5b7e9f4a3345

**Comment by tonybaloney** [comment]
Patch diff file attached to jira ticket: https://issues.apache.org/jira/browse/LIBCLOUD-698

File: docs/compute/_supported_methods_block_storage.rst [hunk 4]
```diff
@@ -9,7 +9,8 @@ Provider                              list volumes create volume destroy volume
 `CloudFrames`_                        no           no            no             no            no            no             no
```
Comment (Kami) [inline docs/compute/_supported_methods_block_storage.rst:9]: Just for the future reference - those documents are updated automatically using `./contrib/generate_provider_feature_matrix_table.py` script.

**Comment by Kami** [comment]
Merged into trunk. Thanks.

**Pull request closed without merging**